python/ray/autoscaler/_private/kuberay/run_autoscaler.py

```diff
@@ -51,17 +62,28 @@ def run_kuberay_autoscaler(cluster_name: str, cluster_namespace: str):
         cluster_name, cluster_namespace
     )
 
-    Monitor(
-        address=ray_address,
-        # The `autoscaling_config` arg can be a dict or a `Callable: () -> dict`.
-        # In this case, it's a callable.
-        autoscaling_config=autoscaling_config_producer,
-        monitor_ip=head_ip,
-        # Let the autoscaler process exit after it hits 5 exceptions.
-        # (See ray.autoscaler._private.constants.AUTOSCALER_MAX_NUM_FAILURES.)
-        # Kubernetes will then restart the autoscaler container.
-        retry_on_failure=False,
-    ).run()
+    gcs_client = GcsClient(ray_address)
+    if is_autoscaler_v2(fetch_from_server=True, gcs_client=gcs_client):
+        from ray.autoscaler.v2.monitor import AutoscalerMonitor as MonitorV2
+
+        MonitorV2(
+            address=gcs_client.address,
+            config_reader=KubeRayConfigReader(autoscaling_config_producer),
+            log_dir=_get_log_dir(),
+            monitor_ip=head_ip,
+        ).run()
+    else:
+        Monitor(
+            address=gcs_client.address,
+            # The `autoscaling_config` arg can be a dict or a `Callable: () -> dict`.
+            # In this case, it's a callable.
+            autoscaling_config=autoscaling_config_producer,
+            monitor_ip=head_ip,
+            # Let the autoscaler process exit after it hits 5 exceptions.
+            # (See ray.autoscaler._private.constants.AUTOSCALER_MAX_NUM_FAILURES.)
+            # Kubernetes will then restart the autoscaler container.
+            retry_on_failure=False,
```

Comment:
Do we need something similar for v2 as well?

Reply:
added comment on the AutoscalerMonitor class.